bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -1139,6 +1139,9 @@ void BCStateTran::setReconfigurationEngineImpl(std::shared_ptr<ClientReconfigura
 template <typename MSG>
 void BCStateTran::freeStateTransferMsg(const MSG *m) {
   const char *p_to_delete = (reinterpret_cast<const char *>(m) - sizeof(MessageBase::Header));
+  if (m->is_incoming_msg_) {
```

Comment:
In this PR I'm aiming at incoming messages only. in the future we might add support for outgoing messages.

We do indeed generate all  incoming messages through the define. However, start transfer messages have a second layer of encapsulation, which causes a problem, since the second layer msg object have no ref of the members of MessageBase related to the buffer. thus, I need to add an is_incoming_msg_ flag to the state transfer messages in order to know whether to update the counters when they release the buffer.